enclave/sgx/start.S

```diff
@@ -13,28 +13,21 @@
 .type _start, @function
 _start:
 .cfi_startproc
+    /* Save the registers passed by EENTER. */
+    pushq %rax
+    pushq %rbx
+    pushq %rcx
+    pushq %rdi
+    pushq %rsi
+    /* Dynamically link liboecore to liboeenclave (install upcalls). */
+    call oe_link_enclave
+    /* Restore the registers passed by EENTER. */
+    popq %rsi
+    popq %rdi
+    popq %rcx
+    popq %rbx
+    popq %rax
```

Comment:
I double-checked that the registers that you saved/restored are passed by EENTER.